src/main/java/com/mraof/minestuck/client/gui/SummonerScreen.java

```diff
@@ -0,0 +1,114 @@
+package com.mraof.minestuck.client.gui;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mraof.minestuck.block.redstone.SummonerBlock;
+import com.mraof.minestuck.entity.MSEntityTypes;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.SummonerPacket;
+import com.mraof.minestuck.tileentity.redstone.SummonerTileEntity;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+
+import java.util.Optional;
+
+public class SummonerScreen extends Screen
+{
+	private static final ResourceLocation guiBackground = new ResourceLocation("minestuck", "textures/gui/generic_medium.png");
+	
+	private static final int guiWidth = 150;
+	private static final int guiHeight = 98;
+	
+	private final SummonerTileEntity te;
+	private boolean isUntriggerable;
+	private int summonRange;
+	
+	public Button incrementButton;
+	public Button decrementButton;
+	private Button unTriggerableButton;
+	
+	private TextFieldWidget entityTypeTextField;
+	
+	
+	SummonerScreen(SummonerTileEntity te)
+	{
+		super(new StringTextComponent("Summoner"));
+		
+		this.te = te;
+		this.summonRange = te.getSummonRange() > 0 ? te.getSummonRange() : 8; //if its defaulted on creation to 0, set it to the intended default of 8
+		this.isUntriggerable = te.getBlockState().getValue(SummonerBlock.UNTRIGGERABLE);
+	}
+	
+	@Override
+	public void init()
+	{
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		addButton(incrementButton = new ExtendedButton(this.width / 2 + 20, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("+"), button -> changeRange(1)));
+		addButton(decrementButton = new ExtendedButton(this.width / 2 - 40, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("-"), button -> changeRange(-1)));
```

Comment:
Since the summon range is in 1-64, it may be nicer to be able to reach a range farther away quicker. I've got two different suggestions for it:
1. Replace the buttons with a text field that the user writes the desired range in
2. Give some means of changing the range by larger amounts with the buttons. I think the most intuitive alternative is to increase the amount if the buttons were clicked with shift held down. Other alternatives include increasing the amount by holding down ctrl, or by using a different mouse button. With the range being 1-64, I think it makes sense for the increased amount to be 8. (So in summary, change the range by 8 when shift-clicking the buttons, change by 1 otherwise)

Reply:
For suggestion 2, would it be something like using the function keyPressed() while in the changeRange function still?

Reply:
I will try to keep this approach in mind for the future, but opted to instead add two buttons that increment or decrement by 10 with the belief that it would be easier to implement and easier for player comprehension